### PR TITLE
chore(storagetransfer): carve out sub-module

### PR DESCRIPTION
**Pull request by codyoss**
This commit will be tagged v0.91.1 and storagetransfer/v0.1.0.

### CODE REVIEW

**Comment by codyoss** [comment]
This PR was generated with the updated code in #4518.
Post merge the following commands should be run:

```bash
git tag -a v0.89.1 <COMMIT-SHA>
git tag -a storagetransfer/v0.1.0 <COMMIT-SHA>
git push origin ref/tags/v0.89.1
git push origin ref/tags/storagetransfer/v0.1.0
```

File: storagetransfer/go_mod_tidy_hack.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// limitations under the License.
+
+// This file, and the cloud.google.com/go import, won't actually become part of
+// the resultant binary.
+// +build modhack
+
+package storagetransfer
+
+// Necessary for safely adding multi-module repo. See: https://github.com/golang/go/wiki/Modules#is-it-possible-to-add-a-module-to-a-multi-module-repository
+import _ "cloud.google.com/go"
```
Comment (noahdietz) [inline storagetransfer/go_mod_tidy_hack.go:22]: Nit: add new line.

**Review by noahdietz — APPROVED** [review]

File: storagetransfer/go_mod_tidy_hack.go (hunk 1 above)
Comment (codyoss) [reply storagetransfer/go_mod_tidy_hack.go:22]: Thanks, missed this. I fixed carver in ac31a180caa4b3660bc3557283fc72b750e7c1d1

**Comment by codyoss** [comment]
These failures are due to local replaces. They will be fixed after tagging, update the dependency, and a tidy.

**Comment by codyoss** [comment]
Will be merging this, later this morning. Checking on a few other things first.

**Pull request merged**